rpmb.c

```diff
@@ -0,0 +1,867 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (C) 2017 Micron Technology Inc.
+ *
+ * Author:
+ *	Bean Huo <beanhuo@micron.com>
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <getopt.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <errno.h>
+
+#include "ufs.h"
+#include "options.h"
+#include "scsi_bsg_util.h"
+#include "rpmb.h"
+#include "hmac_sha/hmac_sha2.h"
+
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+#define DEBUG 1
+static int read_rpmb_status(struct rpmb_opt *opt);
+static int rpmb_read_counter(struct rpmb_opt *opt);
+static int rpmb_program_key(struct rpmb_opt *opt);
+static int rpmb_data_read(struct rpmb_opt *opt);
+static int rpmb_data_write(struct rpmb_opt *opt);
+
+static struct ufs_rpmb_type rpmb_op_t[] = {
+	{"status", RPMB_STATUS, read_rpmb_status},
```

Comment:
what is the nature of this rpmb operation "read status" ? I do not recall seeing anything like that in the spec.
RPMB should only respond to security protocol commands,  Why send request sense?